tests/ci/azure_pipeline_test/release_pipeline.yml

```diff
@@ -0,0 +1,209 @@
+    - unit
+    task_name: "Test - Unit Linux GPU"
+    conda_env: "release_unit_linux_gpu"
+    conda_opts: "python=3.6 cudatoolkit=10.0 \"cudnn>=7.6\""
+    pip_opts: "[gpu] -f https://download.pytorch.org/whl/cu100/torch_stable.html"
+    pytest_markers: "not notebooks and not spark and gpu"
+    install: "release"
+
+- template: dsvm_linux_template.yml
+  parameters:
+    test_types:
+    - unit
+    task_name: "Test - Unit Notebook Linux GPU"
+    conda_env: "release_unit_notebook_linux_gpu"
+    conda_opts: "python=3.6 cudatoolkit=10.0 \"cudnn>=7.6\""
+    pip_opts: "[gpu,examples]  -f https://download.pytorch.org/whl/cu100/torch_stable.html"
+    pytest_markers: "notebooks and not spark and gpu"
+    install: "release"
+
+- template: dsvm_linux_template.yml
+  parameters:
+    test_types:
+    - unit
+    task_name: "Test - Unit Linux Spark"
+    conda_env: "release_unit_linux_spark"
+    conda_opts: "python=3.6"
+    pip_opts: "[spark]"
+    pytest_markers: "not notebooks and spark and not gpu"
+    install: "release"
+
+- template: dsvm_linux_template.yml
+  parameters:
+    test_types:
+    - unit
+    task_name: "Test - Unit Notebook Linux Spark"
+    conda_env: "release_unit_notebook_linux_spark"
+    conda_opts: "python=3.6"
+    pip_opts: "[spark,examples]"
+    pytest_markers: "notebooks and spark and not gpu"
+    install: "release"
+
+# ====== Nightly tests ======
+- template: dsvm_linux_template.yml
+  parameters:
+    test_types:
+    - smoke
+    - integration
+    task_name: "Test - Nightly Linux CPU"
+    timeout: 180 
+    conda_env: "release_nightly_linux_cpu"
+    conda_opts: "python=3.6"
+    pip_opts: "[examples]"
+    pytest_markers: "not spark and not gpu"
+    install: "release"
+
+- template: dsvm_linux_template.yml
+  parameters:
+    test_types:
+    - smoke
+    - integration
+    task_name: "Test - Nightly Linux GPU"
+    timeout: 240
+    conda_env: "release_nightly_linux_gpu"
+    conda_opts: "python=3.6 cudatoolkit=10.0 \"cudnn>=7.6\""
+    pip_opts: "[gpu,examples]  -f https://download.pytorch.org/whl/cu100/torch_stable.html"
+    pytest_markers: "not spark and gpu"
+    install: "release"
+
+- template: dsvm_linux_template.yml
+  parameters:
+    test_types:
+    - smoke
+    - integration
+    task_name: "Test - Nightly Linux Spark"
+    timeout: 180
+    conda_env: "release_nightly_linux_spark"
+    conda_opts: "python=3.6"
+    pip_opts: "[spark,examples]"
+    pytest_markers: "spark and not gpu"
+    install: "release"
+
+# ====== Publish release ======
+- job: Package
+  pool:
+    name: $(Agent_Pool)
+  dependsOn:
+  - release_unit_linux_cpu
+  - release_unit_notebook_linux_cpu
+  - release_unit_linux_gpu
+  - release_unit_notebook_linux_gpu
+  - release_unit_linux_spark
+  - release_unit_notebook_linux_spark
+  - release_nightly_linux_cpu
+  - release_nightly_linux_gpu
+  - release_nightly_linux_spark
+  condition: succeeded()
+  steps:
+    # Create archives with complete source code included
```

Comment:
what are the archives for and do we need separate files for reco_utils and examples? also, do we need both tarballs and zips?

Reply:
being super honest, the only reason is that I had the code of both, but happy to kill the tar one if you guys think it will simplify

Reply:
This is the recommended practice. See https://packaging.python.org/tutorials/packaging-projects/#generating-distribution-archives: "The tar.gz file is a source archive whereas the .whl file is a built distribution. Newer pip versions preferentially install built distributions, but will fall back to source archives if needed. You should always upload a source archive and provide built archives for the platforms your project is compatible with."

Reply:
the tarball mentioned in that doc is already created by the `python setup.py sdist bdist_wheel` command and resides in the dist/ folder next to the wheel. Creating additional archives of the source and example code is unnecessary since we are tagging the release

Reply:
I think we are talking here about two different things, the tar.gz generated by `python setup.py ...` is the package, as Andreas mentioned. It has the same content as the wheel. The tar.gz is a copy of the code at the time of the release, the same as the zip, but with better compression. 

Since the good practice is to upload the wheel and the package tar.gz, we should do that. Also with the code, we can have just the zip for simplicity.

> Creating additional archives of the source and example code is unnecessary since we are tagging the release

I would argue that this improves the user experience, it seems easier for the user to download just the utilities and examples from the zip directly than to the tag, download all the repository and then grab the part it is needed. All this is inspired in LightGBM.

Does it make sense?

Reply:
I don't really see the use case for this. Pulling the auto generated archive from the tag seems like it would be easy enough and pretty standard. It's not that much extra effort on our end but without a specific scenario driving it to me it seems like overkill. Up to you if you want to keep it.

Reply:
ok, maybe I'm wrong and it is easy to download the tag. @anargyri what's your view on this?

Reply:
So, if I understand the discussion, we all agree that the tarball and wheel should be in the archives, So the question is about the zip archive with the source code. I guess we could skip it because people can click on the tagged commit and download it from there as Scott said. It's one extra click more or less. I am a bit curious why LightGBM have it, maybe we could ask them.

Reply:
ok sounds good, I'll remove that part then

Reply:
PR here: https://github.com/microsoft/recommenders/pull/1442